meta-resin-rocko/recipes-utils/aufs-util/aufs-util_%.bbappend

```diff
@@ -0,0 +1 @@
+SECURITY_CFLAGS_pn-${PN} = "${SECURITY_NOPIE_CFLAGS}"
```

Comment:
Is this compiled statically now?

Reply:
I believe it always used to be, you can check a non-rocko build from another board (pi3 for example)

Reply:
So how come we didn't need this configuration until now?

Reply:
https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-containers/tini/tini_0.14.0.bb#L20

Reply:
This is for aufs-util, not tini.

Reply:
Apparently because Poky Rocko added PIE by default. I detailed in the git commit log

Reply:
So currently, we compile with pie as well by default. But not through gcc compilation but through gcc flags. So right now aufs-utils should be compiled with pie. I just checked this on a artik710. They pie flags are passed just fine.

Reply:
So the fix for this is different?

Reply:
I would say that current behavior is broken. If now we compile with PIE also, then we should also be linking with rcrt1.o. And I don't think we do as that object does not exist in Morty either

Reply:
It is not broken as build works and the linker find everything it needs. There is another reason for this breakage. We should investigate.

Reply:
Broken in the sense that the PIE enabling mechanism in Poky Morty is not doing what it should be doing. Not that our builds are broken.

Reply:
Sure. We need to figure out what is broken in this case.

Reply:
So the fix for this error is: either disable PIE for aufs-util or disable PIE for the entire build altogether (i.e. build gcc without PIE).
https://patchwork.openembedded.org/patch/147590/ adds patch 0048-gcc-Enable-static-PIE.patch in which it states that static linking assumes linking with rcrt1.o

1. Can you confirm that even on Poky Morty aufs-util is linked statically? On Poky Rocko it is linked static.
2. Can you confirm that glibc in both Poky Morty, Poky Rocko or any other release lacks rct1.o?

Based on these 2 above, we have 2 options in my opinion:
a - disable PIE for the entire build and drop this PR
b - use this PR

Which option shall we go with? Is there another option?

Reply:
I don't really understand how this fixes the problem. If you deactivate pie on the objects as you do above, that configuration will end up in CFLAGS which will make the objects be compiled without pie. The problem is that later when the gcc is invoked to create the final binary from the objects, there are no pie flags passed and theoretically because of the default configuration now, it should activate pie and fpie and fail in the same way. Do you have any idea why this fixes it?

Reply:
No.
I did not do a deep investigation on gcc for this task. I only looked at how to solve our specific issue withing the confines of poky.
All the conclusions I came up with are here in the git log and in the comments afterwards.